yxi/axi-calyx/axi-combined-calyx.futil

```diff
@@ -446,21 +443,21 @@ component m_awwrite_channel(
       // this contains blocking logic previously in its own group
       group do_aw_transfer {
           //assert AWVALID
-          is_awvalid.in = !awvalid_was_high.out ? 1'b1;
+          is_awvalid.in = !(is_awvalid.out & AWREADY) & !aw_handshake_occurred.out ? 1'b1;
```

Comment:
I think the way to read this is:

> This channel becomes valid whenever the remote side tells us it's ready. But we only do that when we were not previously valid, and we only do the handshake once during the execution of this group.

I guess one small question is: do we need `!is_awvalid.out` as part of this condition? That is, it seems like the condition `AWREADY & !aw_handshake_occurred.out` might suffice, because it will "suppress validity" on every cycle after we are valid for a single cycle.

Reply:
I think this is nice in that it helps point to the idea that `.in` is partitioned by our guards. In other words, it hopefully makes the application of De Morgan's more obvious.  Another thing is that this will make `is_awvalid.in` depend on `AWREADY` in the abstract sense, which goes against the spirit of the recommendation we have [discussed previously](https://cucapra.slack.com/archives/C05TRBNKY93/p1703192144763769). 

| `is_awvalid.out`  | `AWREADY` | `aw_handshake_occurred.out` | Desired input |
| ------------- | -------------      | ------------------------------- | -------------  |
| 0 | 0 | 0 | 1 | 
| 0 | 0 | 1 | 0 |
| 0 | 1 | 0 | 1 |
| 0 | 1 | 1 | 0 |
| 1 | 0 | 0 | 1 |
| 1 | 0 | 1 | 0 |
| 1 | 1 | 0 | 0 |
| 1 | 1 | 1 | 0 |

If the above is correct we care about `is_awvalid.out` when `{AWREADY, aw_handshake_occurred.out}` is `10`. Thinking about this case,  `is_awvalid.out` can probably help us save a cycle, by making `AWVALID` immediately after the handshake occurs, as opposed to waiting 2 cycles for `aw_handshake_occurred.out` to propagate.

Reply:
Aha, I'm pretty sure that makes sense! Thanks for clarifying. The crux for me here was that `AWREADY` gives us the signal to stop the handshake one cycle earlier than `!is_awvalid.out`.